File: FormalBench-data/FilteredData/specifications/seed_726_per_category_10_653ade686f/c/SumNums.c
#include "java_arrays.h"
#include <stdint.h>

/*@
  assigns \nothing;
  ensures \result == (int32_t)((integer)left + right);
*/
static int32_t java_add(int32_t left, int32_t right) {
    return (int32_t)((uint32_t)left + (uint32_t)right);
}

/*@
  assigns \nothing;
  ensures \result == (m <= (int32_t)((integer)x + y) && (int32_t)((integer)x + y) <= n ? 20 : (int32_t)((integer)x + y));
*/
int32_t sumNums(int32_t x, int32_t y, int32_t m, int32_t n) {
    int32_t sum = java_add(x, y);
    if (sum >= m && sum <= n) {
        return 20;
    }
    return sum;
}

File: FormalBench-data/FilteredData/specifications/seed_726_per_category_10_653ade686f/java/SumNums.java
import java.io.*;
import java.lang.*;
import java.util.*;
import java.math.*;


class SumNums {
    
    /*@ public normal_behavior
      @ assignable \nothing;
      @ ensures \result == (m <= \java_math(x + y) && \java_math(x + y) <= n ? 20 : \java_math(x + y));
      @*/
    //@ code_java_math
    public static int sumNums(int x, int y, int m, int n) {
        int sum = x + y;
        if (sum >= m && sum <= n) {
            return 20;
        } else {
            return sum;
        }
    }
}

